### PR TITLE
Rename input.application.workloads to input.application.prometheus

### PR DESCRIPTION
**Pull request by skhalash**
## Description

Changes proposed in this pull request:

- As mentioned in [the comment](https://github.com/kyma-project/kyma/issues/17828#issuecomment-1647757217), the "Workloads" application input source has to be renamed into "Prometheus"

## Related Issues and Documents

Related issues:
https://github.com/kyma-project/kyma/issues/17828

## Traceability

- [x] The PR is linked to a GitHub Issue.
- [ ] New features have a milestone label set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub Issue has a respective `area` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:

- [x] Yes.
- [ ] No, because unit tests are not needed.
- [ ] No, because of ...

The feature is e2e-tested:

- [x] Yes.
- [ ] No, because e2e-tests are not needed.
- [ ] No, because of ...

Tests conducted for the PR:

## Codebase

- [x] My code follows the style guidelines of this project.
- [x] The code was planned and designed following the defined architecture and the separation of concerns.
- [ ] The code has sufficient comments, particularly for all hard-to-understand areas.
- [x] This PR adds value and shows no feature creep.
- [x] I have augmented the test suite that proves my fix is effective or that my feature works.
- [x] Adjusted the documentation if the change is user-facing.

### CODE REVIEW

**Review by hisarbalik — APPROVED** [review]

File: docs/user/resources/05-metricpipeline.md [hunk 1]
```diff
@@ -51,10 +51,10 @@ For details, see the [MetricPipeline specification file](https://github.com/kyma
 | ---- | ----------- | ---- |
 | **input**  | object | Configures different inputs to send additional metrics to the metric gateway. |
 | **input.&#x200b;application**  | object | Configures application related scraping. |
+| **input.&#x200b;application.&#x200b;prometheus**  | object | Configures Prometheus scraping. |
+| **input.&#x200b;application.&#x200b;prometheus.&#x200b;enabled**  | boolean | If enabled, Pods marked with prometheus.io/scrape=true annotation will be scraped. |
```
Comment (IwonaLanger) [inline docs/user/resources/05-metricpipeline.md:55]:
```suggestion
| **input.&#x200b;application.&#x200b;prometheus.&#x200b;enabled**  | boolean | If enabled, Pods marked with `prometheus.io/scrape=true` annotation will be scraped. |
```

Comment (skhalash) [reply docs/user/resources/05-metricpipeline.md:55]: Unfortunately, adding ticks, quotes, etc. is a bit problematic. The README is generated from a CRD definition (YAML file), which, in turn, is generated from a Go struct.

**Review by IwonaLanger — APPROVED** [review]

File: docs/user/resources/05-metricpipeline.md (hunk 1 above)
Comment (skhalash) [reply docs/user/resources/05-metricpipeline.md]: Fixed

**Review by hisarbalik — APPROVED** [review]

**Pull request merged**